tests/page/locator-query.spec.ts

```diff
@@ -59,3 +59,23 @@ it('should throw on due to strictness 2', async ({ page }) => {
   const e = await page.locator('option').evaluate(e => {}).catch(e => e);
   expect(e.message).toContain(`strict mode violation`);
 });
+
+it('should filter by text', async ({ page }) => {
```

Comment:
Let's add a test with `<div>foo<span>hello world</span>bar</div>` and `locator('div').withText('hello world')`?

Reply:
I hope this test would fail? We are interested in the outermost node matching the text here.